networkx/algorithms/lowest_common_ancestors.py

```diff
@@ -10,9 +10,121 @@
     "all_pairs_lowest_common_ancestor",
     "tree_all_pairs_lowest_common_ancestor",
     "lowest_common_ancestor",
+    "naive_lowest_common_ancestor",
+    "naive_all_pairs_lowest_common_ancestor",
 ]
 
 
+def _get_a_lowest_common_ancestor(G, common_ancestors):
+    """Return a lowest common ancestor from the given set of common ancestors"""
+    # Start with an arbitrary node v from the set of common ancestors.  Follow
+    # arbitrary outgoing edges, remaining in the set of common ancestors, until
+    # reaching a node with no outgoing edge to another of the common ancestors.
+
+    v = next(iter(common_ancestors))
+    while True:
+        successor = None
+        for w in G.successors(v):
+            if w in common_ancestors:
+                successor = w
+                break
+        if successor is None:
+            return v
+        v = successor
```

Comment:
Perhaps this function could be "in-lined". Is it likely to be used again for another function?  If so leave it where it is. 

Also, see what you think of this approach to the same looping structure. 
```
    v = next(iter(common_ancestors))
    while True:
        try:
            v = next(w for w in G.successors(v) if w in common_ancestors)
        except StopIteration:
            return v
```
There is also a python for/else construction you could use here. The else clause is only executed if the for loop completes without a "break".  Notice that the `else` is indented with the `for` and not with the `if`.
```
    v = next(iter(common_ancestors))
    while True:
        for w in G.successors(v):
            if w in common_ancestors:
                v = w
                break
        else:
            return v
```

All of these are the same, so don't worry about changing it, but you might like thinking of these ways and the flexibility of python and whether one is more beautiful in your estimation.